pkg/river/internal/value/decode_test.go

```diff
@@ -663,31 +664,42 @@ func TestDecode_KnownTypes_Any(t *testing.T) {
 		input  any
 		expect any
 	}{
-		// All numbers must decode to float64.
-		{int(15), float64(15)},
-		{int8(15), float64(15)},
-		{int16(15), float64(15)},
-		{int32(15), float64(15)},
-		{int64(15), float64(15)},
-		{uint(15), float64(15)},
-		{uint8(15), float64(15)},
-		{uint16(15), float64(15)},
-		{uint32(15), float64(15)},
-		{uint64(15), float64(15)},
+		// expect "int"
+		{int(0), int(0)},
+		{int(-1), int(-1)},
+		{int(15), int(15)},
+		{int8(15), int(15)},
+		{int16(15), int(15)},
+		{int32(15), int(15)},
+		{int64(15), int(15)},
+		{uint(0), int(0)},
+		{uint(15), int(15)},
+		{uint8(15), int(15)},
+		{uint16(15), int(15)},
+		{uint32(15), int(15)},
+		{uint64(15), int(15)},
+		{int64(math.MinInt64), int(math.MinInt64)},
+		{int64(math.MaxInt64), int(math.MaxInt64)},
+		// expect "uint"
+		{uint64(math.MaxInt64 + 1), uint64(math.MaxInt64 + 1)},
+		{uint64(math.MaxUint64), uint64(math.MaxUint64)},
+		// expect "float"
 		{float32(2.5), float64(2.5)},
 		{float64(2.5), float64(2.5)},
+		{float64(math.MinInt64) - 10, float64(math.MinInt64) - 10},
+		{float64(math.MaxInt64) + 10, float64(math.MaxInt64) + 10},
 
 		{bool(true), bool(true)},
 		{string("Hello"), string("Hello")},
 
 		{
 			input:  []int{1, 2, 3},
-			expect: []any{float64(1), float64(2), float64(3)},
+			expect: []any{int(1), int(2), int(3)},
```

Comment:
I don't think the wrapping is needed here anymore, since I'm pretty sure untyped whole number constants default to `int`:

```suggestion
			expect: []any{1, 2, 3},
```

This change can be replicated in all other files where float64(N) is changed to int(N); it should be possible to remove the explicit type cast, which is half the reason to support `int` anyway.